incubator/binding-kafka-grpc/src/main/java/io/aklivity/zilla/runtime/binding/kafka/grpc/internal/stream/KafkaGrpcRemoteServerFactory.java

```diff
@@ -0,0 +1,1994 @@
+
+            assert acknowledge <= sequence;
+
+            initialAck = acknowledge;
+            initialMax = maximum;
+            state = KafkaGrpcState.openInitial(state);
+
+            assert initialAck <= initialSeq;
+
+            doKafkaTombstone(traceId, authorization, lastCorrelationId);
+        }
+
+        private void onKafkaReset(
+            ResetFW reset)
+        {
+            final long sequence = reset.sequence();
+            final long acknowledge = reset.acknowledge();
+            final long traceId = reset.traceId();
+            final long authorization = reset.authorization();
+
+            assert acknowledge <= sequence;
+
+            state = KafkaGrpcState.closeInitial(state);
+
+            doKafkaReset(traceId, authorization);
+        }
+
+        private void doKafkaReset(
+            long traceId,
+            long authorization)
+        {
+            if (KafkaGrpcState.replyOpened(state) &&
+                !KafkaGrpcState.replyClosed(state))
+            {
+                state = KafkaGrpcState.closeReply(state);
+
+                doReset(kafka, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                    traceId, authorization, emptyRO);
+            }
+        }
+
+        private void doKafkaWindow(
+            long traceId)
+        {
+            if (kafka != null && !KafkaGrpcState.replyClosed(state))
+            {
+                doWindow(kafka, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                    traceId, 0L, replyBud, replyPad, replyCap);
+            }
+        }
+
+        private void doKafkaTombstone(
+            long traceId,
+            long authorization,
+            OctetsFW correlationId)
+        {
+            if (!KafkaGrpcState.initialOpened(state))
+            {
+                doKafkaBegin(traceId, authorization, 0);
+                lastCorrelationId =  new OctetsFW.Builder()
+                    .wrap(new UnsafeBuffer(new byte[correlationId.sizeof()]), 0, correlationId.sizeof())
+                    .set(correlationId)
+                    .build();
+            }
+            else if (correlationId != null)
+            {
+                Flyweight tombstoneDataEx = kafkaDataExRW
+                    .wrap(extBuffer, 0, extBuffer.capacity())
+                    .typeId(kafkaTypeId)
+                    .merged(m -> m
+                        .timestamp(now().toEpochMilli())
+                        .partition(p -> p.partitionId(-1).partitionOffset(-1))
+                        .key(k -> result.key(correlationId, k))
+                        .headers(h -> result.headersWithStatusCode(correlationId, HEADER_VALUE_GRPC_INTERNAL_ERROR, h)))
+                    .build();
+
+                doKafkaData(traceId, authorization, 0, 0, DATA_FLAG_COMPLETE, null, tombstoneDataEx);
+                lastCorrelationId = null;
+            }
+        }
+    }
+
+    private final class KafkaFetchRemote
```

Comment:
Suggest renaming this to `KafkaRemoteServer` and unifying with `RemoteServer` above.

Reply:
Makes sense.